markdown/developers/test_data_guidelines.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Test data guidelines
+subtitle: Guidelines for adding test data to nf-core repositories
+---
+
+## Overview
+
+The nf-core community currently hosts two sets of test data that are used to execute CI tests and validate results: pipelines and modules
+
+For consistency and keeping the test data repositories small, we provide the following guidelines and recommendations that developers should aim to follow when adding new data.
+
+The general philosophy to keep in mind for all new test data is: **as small as possible, as large as necessary**.
+
+## Modules
+
+### Re-use existing test data
+
+Use what already exists! The priority is use in your module tests whatever is already available on the `modules` branch of the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
+
+If a test data file for your module can be be _quickly_ generated by an upstream module, don't add the input file to the test data. Run the upstream module in the test for your module.
+For example, if your module requires the generate of a very small index file, include the indexing module in the test `main.nf` file, and pass the output channel of the index to the new module.
+
+In contrast, if your module requires many tests, steps, or heavy CPU usage, you _should_ store such additional files on the test datasets repository (see below).
+
+### Adding new test data
+
+1. Always ask on the [#modules](https://nfcore.slack.com/channels/modules) or [#test-data](https://nfcore.slack.com/channels/modules) channels on the [nf-core slack](https://nf-co.re/join/slack) before adding test data.
+
+2. Add any new test data from a pull request from a personal fork of the nf-core/test-dataset repository.
+
+3. New bioinformatic test data files should be generated from the existing collections of test data as far as possible. See field-specific guidance [below](#field-specific-guidance).
+
+   For example, if you were adding new genomic test data, and needed to create a specific type of genotyping file format for a particular tool - you should use the SARS-CoV2 or _Homo sapiens_ BAM files as input when generating the file you will need to use your module tests. The resulting genotyping file should then be stored alongside the BAM file, or in the corresponding file format type directory.
+
+4. In the worst case scenario where you cannot (re)use the exisiting data in the field-specific collections, the `delete_me/` folder may be used.
+
+   - We discourage the use of this directory as far as possible.
+   - If you do use this directory, we ask you to replace anything added to this directory with proper test data based on the existing dataset as soon as possible.
+
+5. For non-bioinformatic specific files (like simple text files, or tables), you can place these in the `generic/` directory.
+
+6. Files must be small (less than the GitHub file-size limit as a maximum size), and must be aggressively subsampled if necessary.
```

Comment:
Maybe mention what the GitHub file size limit is, or don't mention it at all. Again I can see people wondering what to aim for and then using this as a sort of target...